### PR TITLE
add post comment functionality

### PR DESCRIPTION
**Pull request by lilyha1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -57,4 +57,44 @@ describe("2. GET /api/reviews", () => {
...
+      });
+  });
+  test("status: 404, msg: resource not found - invalid review_id", () => {
+    const comment = { username: "dav3rid", body: "could be better" };
+    return request(app)
+      .post("/api/reviews/10000/comments")
+      .send(comment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("review_id: 10000 not found");
```
Comment (mezzDavies) [inline __tests__/app.test.js]: This is giving away a column name in the error message - we need to be careful about giving out information about databases - "not found" would be better practice.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -57,4 +57,44 @@ describe("2. GET /api/reviews", () => {
...
+          comment_id: 7,
+          votes: 0,
+          created_at: expect.any(String),
+          author: comment.username,
+          body: comment.body,
+          review_id: 3,
+        });
+      });
+  });
+  test("status: 404, msg: resource not found - invalid review_id", () => {
```
Comment (mezzDavies) [inline __tests__/app.test.js]: This test is good but is actually testing for a **valid** but non-existant review ID. 1000 is a valid id but it just doesn't exist.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -57,4 +57,44 @@ describe("2. GET /api/reviews", () => {
...
+  test("status: 404, invalid user", () => {
+    const comment = { username: "hlily", body: "could be better" };
+    return request(app)
+      .post("/api/reviews/3/comments")
+      .send(comment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("username: hlily not found");
+      });
+  });
```
Comment (mezzDavies) [inline __tests__/app.test.js:194]: We also need some further tests here as we need to:

- handle if a bad (invalid) review_id is passed eg /api/reviews/:"not-an-id"/comments
- handle if required fields are missing from the request body

Make sure you consider the error codes too.

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Looking good so far but you need some further tests and handling of possible errors for this endpoint. (Also please be careful your test description matches what the test actually tests for eg error codes etc.)

**Review by MKD01 — APPROVED** [review]
Looks good, happy for this to be merged 👍

**Pull request merged**